diff -ru a/src/gcc/gcc/config/arm/arm-c.c b/src/gcc/gcc/config/arm/arm-c.c
--- a/src/gcc/gcc/config/arm/arm-c.c	2016-02-24 04:19:12.000000000 -0400
+++ b/src/gcc/gcc/config/arm/arm-c.c	2016-08-13 16:55:03.509143808 -0300
@@ -189,6 +189,8 @@
   builtin_assert ("cpu=arm");
   builtin_assert ("machine=arm");
 
+  builtin_define ("__vita__");
+
   arm_cpu_builtins (pfile);
 }
 
diff -ru a/src/gcc/gcc/gcc.c b/src/gcc/gcc/gcc.c
--- a/src/gcc/gcc/gcc.c	2016-02-19 18:18:38.000000000 -0400
+++ b/src/gcc/gcc/gcc.c	2016-08-13 17:13:54.104949572 -0300
@@ -658,8 +658,9 @@
 #endif
 
 /* config.h can define LIB_SPEC to override the default libraries.  */
+#undef LIB_SPEC
 #ifndef LIB_SPEC
-#define LIB_SPEC "%{!shared:%{g*:-lg} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}}"
+#define LIB_SPEC "%{!shared:%{g*:-lc} %{!p:%{!pg:-lc}}%{p:-lc_p}%{pg:-lc_p}} -lSceRtc_stub -lSceSysmem_stub -lSceKernelThreadMgr_stub -lSceKernelModulemgr_stub -lSceIofilemgr_stub -lSceProcessmgr_stub -lSceLibKernel_stub -lSceNet_stub"
 #endif
 
 /* When using -fsplit-stack we need to wrap pthread_create, in order
 
diff -ru a/src/gcc/libstdc++-v3/configure b/src/gcc/libstdc++-v3/configure
--- a/src/gcc/libstdc++-v3/configure
+++ b/src/gcc/libstdc++-v3/configure
@@ -5369,13 +5369,6 @@ $as_echo "$as_me: OS config directory is $os_include_dir" >&6;}
 
 
 
-# Libtool setup.
-if test "x${with_newlib}" != "xyes"; then
-  enable_dlopen=yes
-
-
-
-fi
 case `pwd` in
   *\ * | *\    *)
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Libtool does not cope well with whitespace in \`pwd\`" >&5
 
diff -ru a/src/gcc/libstdc++-v3/configure b/src/gcc/libstdc++-v3/configure
--- a/src/gcc/libstdc++-v3/configure
+++ b/src/gcc/libstdc++-v3/configure
@@ -53741,7 +53741,7 @@ $as_echo "#define HAVE_TLS 1" >>confdefs.h
        ;;
     esac
     ;;
-  *-linux* | *-uclinux* | *-gnu* | *-kfreebsd*-gnu | *-cygwin*)
+  *-linux* | *-uclinux* | *-gnu* | *-kfreebsd*-gnu | *-cygwin* | arm-vita-eabi*)
 
   # All these tests are for C++; save the language and the compiler flags.
   # The CXXFLAGS thing is suspicious, but based on similar bits previously

diff -ru a/src/gcc/libstdc++-v3/include/std/type_traits b/src/gcc/libstdc++-v3/include/std/type_traits
--- a/src/gcc/libstdc++-v3/include/std/type_traits
+++ b/src/gcc/libstdc++-v3/include/std/type_traits
@@ -38,7 +38,7 @@
 #include <bits/c++config.h>
 
 #ifdef _GLIBCXX_USE_C99_STDINT_TR1
-# if defined (__UINT_LEAST16_TYPE__) && defined(__UINT_LEAST32_TYPE__)
+# if defined (__UINT_LEAST16_TYPE__) && defined(__UINT_LEAST32_TYPE__) && 0
 namespace std
 {
   typedef __UINT_LEAST16_TYPE__ uint_least16_t;

diff -ru a/src/gcc/gcc/config/arm/unknown-elf.h b/src/gcc/gcc/config/arm/unknown-elf.h
--- a/src/gcc/gcc/config/arm/unknown-elf.h
+++ b/src/gcc/gcc/config/arm/unknown-elf.h
@@ -29,7 +29,7 @@
 #endif
 
 /* Now we define the strings used to build the spec file.  */
-#define UNKNOWN_ELF_STARTFILE_SPEC	" crti%O%s crtbegin%O%s crt0%O%s"
+#define UNKNOWN_ELF_STARTFILE_SPEC	" crti%O%s crtbegin%O%s crt1%O%s"
 
 #undef  STARTFILE_SPEC
 #define STARTFILE_SPEC	\


